src/Illuminate/Support/Number.php

```diff
@@ -141,30 +141,30 @@ public static function fileSize(int|float $bytes, int $precision = 0, ?int $maxP
      * @param  int|null  $maxPrecision
      * @return string
      */
-    public static function forHumans(int|float $number, int $precision = 0, ?int $maxPrecision = null)
+    public static function forHumans(int|float $number, int $precision = 0, ?int $maxPrecision = null, ?string $locale = null)
     {
-        $units = [
-            3 => 'thousand',
-            6 => 'million',
-            9 => 'billion',
-            12 => 'trillion',
-            15 => 'quadrillion',
-        ];
-
         switch (true) {
             case $number === 0:
                 return '0';
             case $number < 0:
-                return sprintf('-%s', static::forHumans(abs($number), $precision, $maxPrecision));
+                return sprintf('-%s', static::forHumans(abs($number), $precision, $maxPrecision, $locale));
             case $number >= 1e15:
-                return sprintf('%s quadrillion', static::forHumans($number / 1e15, $precision, $maxPrecision));
+                return sprintf('%s %s', static::forHumans($number = $number / 1e15, $precision, $maxPrecision, $locale), trans_choice('quadrillion', $number, locale: $locale ?? static::$locale));
         }
 
         $numberExponent = floor(log10($number));
         $displayExponent = $numberExponent - ($numberExponent % 3);
         $number /= pow(10, $displayExponent);
 
-        return trim(sprintf('%s %s', static::format($number, $precision, $maxPrecision), $units[$displayExponent] ?? ''));
+        $units = [
+            3 => trans_choice('thousand', $number, locale: $locale ?? static::$locale),
```

Comment:
Is it right to use global helper methods inside of the framework?

Reply:
Good question! I think it's alright as I see global translation helpers being used in Framework code like validation and pagination helpers. But let's keep this open until we get input from someone from the team!